pkg/cloudprovider/providers/vsphere/vclib/virtualmachine.go

```diff
@@ -400,3 +401,10 @@ func (vm *VirtualMachine) deleteController(ctx context.Context, controllerDevice
 	}
 	return nil
 }
+
+// RenewVM renews this virtual machine with new client connection.
+func (vm *VirtualMachine) RenewVM(client *govmomi.Client) VirtualMachine {
+	dc := Datacenter{Datacenter: object.NewDatacenter(client.Client, vm.Datacenter.Reference())}
+	newVM := object.NewVirtualMachine(client.Client, vm.VirtualMachine.Reference())
+	return VirtualMachine{VirtualMachine: newVM, Datacenter: &dc}
```

Comment:
I think it would be better to update same virtual machine object instead of creating the new one. In this case, the update should be mutually exclusive.

Reply:
That approach has side effects. It would be difficult to account for all possible outcomes for cases where one thread has access to previous VM and this method simply changes it in background. Note that those two threads won't be using same lock to prevent any possible race condition.

Reply:
That approach -> The one that you are suggesting...